File: app/src/main/java/br/com/ricardo/filmespopulares/data/network/api/ApiService.java
package br.com.ricardo.filmespopulares.data.network.api;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class ApiService {

    private static FilmService service;

    public static FilmService getInstance(){

        if(service == null){

            Retrofit retrofit = new Retrofit.Builder()
                    .baseUrl(FilmService.BASE_URL)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build();

            service = retrofit.create(FilmService.class);

        }

        return service;

    }
}

File: app/src/main/java/com/brunoalmeida/movies/data/APIService.kt
package com.brunoalmeida.movies.data

import com.brunoalmeida.movies.BuildConfig
import retrofit2.Retrofit
import retrofit2.converter.moshi.MoshiConverterFactory

object APIService {

    private fun initRetrofit(): Retrofit {
        return Retrofit.Builder()
            .baseUrl(BuildConfig.BASE_URL)
            .addConverterFactory(MoshiConverterFactory.create())
            .build()
    }

    val service: MDBServices = initRetrofit().create(MDBServices::class.java)

}